bindings/java/src/org/sleuthkit/datamodel/Report.java

```diff
@@ -39,15 +39,16 @@
 
 	static long ID_NOT_SET = -1;
 	private long objectId = ID_NOT_SET;
-	private final Path path;
+	private final String pathAsString;
+	private final Path pathAsPath; // NULL if path is for a URL
```

Comment:
I don't see the value in having two fields here. It just complicates the code. A string path will be cleaner. I would not make the implicit isURL flag explicit either. The small cost of a path.startsWith("http") here and there does not justify adding a flag field, in my opinion.